fix: Correct AdminUser type definition to use camelCase

@claude:

Objective:
Fix the TypeScript errors in AdminPage.tsx by correcting the AdminUser type definition.

Build Error:

error TS2551: Property 'createdAt' does not exist on type 'AdminUser'. Did you mean 'created_at'?

error TS2551: Property 'lastSignInAt' does not exist on type 'AdminUser'. Did you mean 'last_sign_in_at'?

Required Fix:

Open the file where the AdminUser type is defined (likely src/lib/types.ts).

In the AdminUser interface, change the property created_at to createdAt.

In the AdminUser interface, change the property last_sign_in_at to lastSignInAt.

Verification:
The build must succeed.

--- a/src/pages/AdminPage.tsx
+++ b/src/pages/AdminPage.tsx
@@ -145,10 +145,10 @@ export default function AdminPage() {
                     </span>
                   </td>
                   <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
-                    {formatDate(user.created_at)}
+                    {formatDate(user.createdAt)}
                   </td>
                   <td className="px-6 py-4 whitespace-nowrap text-sm text-gray-500">
-                    {user.last_sign_in_at ? formatDate(user.last_sign_in_at) : 'Never'}
+                    {user.lastSignInAt ? formatDate(user.lastSignInAt) : 'Never'}
                   </td>
                   <td className="px-6 py-4 whitespace-nowrap text-sm">
                     <select

--- a/src/services/adminService.ts
+++ b/src/services/adminService.ts
@@ -7,8 +7,8 @@ export interface AdminUser {
   email: string
   username: string
   role: UserRole
-  created_at: string
-  last_sign_in_at?: string
+  createdAt: string
+  lastSignInAt?: string
 }
 
 export class AdminService {
@@ -82,7 +82,16 @@ export class AdminService {
         throw new Error(data.error)
       }
 
-      return data?.users || []
+      // Transform snake_case to camelCase for frontend compatibility
+      const users = data?.users || []
+      return users.map((user: any) => ({
+        id: user.id,
+        email: user.email,
+        username: user.username,
+        role: user.role,
+        createdAt: user.created_at,
+        lastSignInAt: user.last_sign_in_at
+      }))
     } catch (error) {
       console.error('Error listing users:', error)
       throw new Error('Failed to list users')

